test/extended/node/minimum_kubelet_version.go

```diff
@@ -0,0 +1,273 @@
+package node
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/onsi/ginkgo/v2"
+	g "github.com/onsi/ginkgo/v2"
+	o "github.com/onsi/gomega"
+	operatorv1client "github.com/openshift/client-go/operator/clientset/versioned"
+	"github.com/openshift/origin/pkg/test/ginkgo/result"
+	exutil "github.com/openshift/origin/test/extended/util"
+	authorizationv1 "k8s.io/api/authorization/v1"
+	v1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	restclient "k8s.io/client-go/rest"
+	"k8s.io/kubernetes/test/e2e/framework"
+)
+
+const (
+	nodesGroup          = "system:nodes"
+	nodeNamePrefix      = "system:node:"
+	desiredTestDuration = 25 * time.Minute
+)
+
+var _ = g.Describe("[sig-node][OCPFeatureGate:MinimumKubeletVersion] ", func() {
+	defer g.GinkgoRecover()
+
+	oc := exutil.NewCLIWithoutNamespace("minimum-kubelet-version")
+
+	g.DescribeTable("admission", func(version string, expectedErr bool) {
+		defer updateMinimumKubeletVersion(oc, version, expectedErr)
+	},
+		g.Entry("should allow an empty minimum kubelet version", "", false),
+		g.Entry("should allow an old minimum kubelet version", "1.30.0", false),
+		g.Entry("should not allow with a new minimum kubelet version", "1.100.0", true),
+	)
+})
+
+type minimumKubeletVersionAuthTestCase struct {
+	testName       string
+	kubeletVersion string
+	testFunc       func()
+}
+
+var _ = g.Describe("[sig-node][OCPFeatureGate:MinimumKubeletVersion] [Serial]", func() {
+	defer g.GinkgoRecover()
+
+	var (
+		nodeImpersonatingClient clientset.Interface
+		nodeName                = "fakenode"
```

Comment:
It may be worth having a random value for fakenode so we don't get conflicts.

You do clean it up so we may not get any issues.

Reply:
yeah honestly I feel like a conflict would be better because that means we'd catch if the test didn't clean up after itself or is suddenly not running serially